vertica/datadog_checks/vertica/vertica.py

```diff
@@ -92,6 +92,37 @@ def __init__(self, name, init_config, instances):
         # Cache database results for re-use among disparate functions
         self._view = defaultdict(list)
 
+        self._metric_groups = {
+            "licenses": self.query_licenses,
+            "license_audits": self.query_license_audits,
+            "system": self.query_system,
+            "nodes": self.query_nodes,
+            "projections": self.query_projections,
+            "projection_storage": self.query_projection_storage,
+            "storage_containers": self.query_storage_containers,
+            "host_resources": self.query_host_resources,
+            "query_metrics": self.query_query_metrics,
+            "resource_pool_status": self.query_resource_pool_status,
+            "disk_storage": self.query_disk_storage,
+            "resource_usage": self.query_resource_usage,
+            "version": self.query_version,
+            "custom": self.query_custom,
+        }
+
+        include_groups = self.instance.get("metric_groups", [])
+
+        # License query needs to be run before getting system
+        if 'system' in include_groups and 'licenses' not in include_groups:
+            include_groups.insert(0, 'licenses')
+
+        self._desired_groups = []
+
+        for group in include_groups:
+            if group in self._metric_groups:
+                self._desired_groups.append(group)
+            else:
+                self.log.debug("Metric group name `%s` is not valid, ignoring metric group", group)
```

Comment:
I think this validation should be done at `__init__` and throw a ConfigurationError if an invalid group is added since it's a mistake on the user's side.